### ISSUE TITLE
SortedMap/TreeMap.iterableFactory should explicitly return Seq instead of Iterable

### ISSUE THREAD

**Issue by texasbruce** [scala/bug#12061]
## reproduction steps

using Scala 2.13.2

```scala
scala> collection.mutable.TreeMap(0->0).iterableFactory
val res0: scala.collection.IterableFactory[scala.collection.mutable.Iterable] = scala.collection.mutable.Iterable$@563843f1

scala> collection.immutable.TreeMap(0->0).iterableFactory
val res1: scala.collection.IterableFactory[scala.collection.immutable.Iterable] = scala.collection.immutable.Iterable$@3a988299
```

## problem

Both mutable and immutable `TreeMap`/`SortedMap`'s `iterableFactory` return just a normal `Iterable`. Although it currently works, because `IterableFactory` is a Delegate of `List`, it is 

- conceptually incorrect. They are sorted in order, and iterableFactory should retain the sequential order. `Iterable` does not guarrantee the sorted order
- if `Iterable` implemenetation changes in the future and no longer `List`, this will break.

Furthermore, currently `keySet` signature returns a `SortedSet`, which is accurate, but `values` returns a simple `Iterable`. It would be more preferred to return a `Seq` instead, so the values would also retain the order same as in the `SortedMap`

**Comment by NthPortal** [comment]
I'm not sure I agree with the premise here. `Seq` and `Sorted` are different properties - the former is insertion order, and the latter follows some `Ordering`. as evidence, see the existence of both `SeqMap` and `SortedMap`. I would agree that `iterableFactory` for `SeqMap` probably ought to return an `IterableFactory[Seq]`, however.

**Comment by texasbruce** [comment]
> `Seq` and `Sorted` are different properties - the former is insertion order, and the latter follows some `Ordering`. 

Not necessarily. `Seq` ensures a sequential order. It could be insertion order, order by `Ordering`, or any non-random order in sequence. (However it does not keep the order if it is modified). My point in this case is, `SortedMap` iterableFactory should not return an Iterable, because it does not inherit the sorted order as in the SortedMap. For example,

    SortedMap("first" -> 2, "second" -> 1).values

return `Iterable[Int]` (it actually returns List because Iterable uses List as underlying collection). The access order would be `(2,1)`, or `(1,2)` but in this case I am looking for `(2,1)` because it is supposed to be sorted by the key and keep that order when I get the values.

**Comment by NthPortal** [comment]
After thinking about this a bunch, I think I agree. Unfortunately, I don't think it would be binary compatible to add overrides to the traits, so I think you'd have to do it in the individual implementations
